### DISCUSSION TITLE
Haskell Language Server version selection?

### DISCUSSION THREAD

**Discussion by levans** [zed-industries/zed#8160] [Feature Requests]
Just stumbled on Zed and I shall be paying attention to this editor (and trying to make it work in my day-to-day too).

However, I mostly program in Haskell.  I have therefore added this to me settings:
"language_servers": {
    "haskell": {
      "command": "/Users/luke/.ghcup/hls/2.6.0.0/bin/haskell-language-server-wrapper",
      "args": [
        "--lsp"
      ],
      "file_types": [
        "hs",
        "lhs"
      ]
    }
  }

The problem is that there are different versions of HLS for different compiler versions.  
I think the VSCode plugin will automatically choose the right HLS and is able to find it in the ghcup locations where they are typically installed. 

I'm guessing that something like that is not currently possible with Zed... but thought I'd ask, just in case :)

**Comment by Moshyfawn** [comment]
You're right, this is not supported in Zed yet. You might find https://github.com/zed-industries/zed/issues/7902#issuecomment-1957415352 useful. Basically the idea is to offload the language server configuration to specific language extensions.

**Reply by levans** [reply]
Thanks for the confirmation.  I shall keep an eye on progress.

**Comment by 4ad** [comment]
So I understand that the plan is to move the HLS version selection in the Haskell extension, but is there a chance that this will ever happen soon?

Currently Zed loads some random HLS instead of the correct one, and this is not workable in Haskell.
